app-backend/db/src/test/scala/com/azavea/rf/database/TeamDaoSpec.scala

```diff
@@ -104,6 +104,28 @@ class TeamDaoSpec extends FunSuite with Matchers with Checkers with DBTestConfig
     }
   }
 
+  test("creating a team with role") {
+    check {
+      forAll (
+        (userCreate: User.Create, orgCreate: Organization.Create, teamCreate: Team.Create) => {
+          val createTeamIO = for {
+            orgAndUserInsert <- insertUserAndOrg(userCreate, orgCreate)
+            (orgInsert, userInsert) = orgAndUserInsert
+            teamInsert <- TeamDao.createWithRole(fixupTeam(teamCreate, orgInsert, userInsert), userInsert)
+            users <- TeamDao.listMembers(teamInsert.id, PageRequest(0, 30, Map.empty), SearchQueryParameters(), userInsert)
```

Comment:
I would also add something like `userIsTeamAdmin <- TeamDao.userIsAdmin(userInsert, teamInsert.id)` to check if the added user has the admin access to the created team since this is a part of this function's feature that I think would be necessary to have it tested.